### ISSUE TITLE
`default_sysctls` setting in `containers.conf` not compatible with `podman run network='host'`

### ISSUE THREAD

**Issue by polygamma** [containers/podman#8482]
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind bug

**Description**

Arch Linux started to include the `containers.conf` as it is which broke `podman run network='host'`.

See: https://bugs.archlinux.org/task/68742

```
default_sysctls = [
 "net.ipv4.ping_group_range=0 0",
]
```
of the `containers.conf` leads to a failing execution of `podman run` with `network='host'`:
```bash
[jonny@jonny-arch-pc ~]$ sudo podman run --network='host' docker.io/hello-world
Error: sysctl "net.ipv4.ping_group_range" is not allowed in the hosts network namespace: OCI runtime error
```

Removing of `"net.ipv4.ping_group_range=0 0"` fixes the problem:
```bash
[jonny@jonny-arch-pc ~]$ sudo podman run --network='host' docker.io/hello-world

Hello from Docker!
This message shows that your installation appears to be working correctly.

To generate this message, Docker took the following steps:
 1. The Docker client contacted the Docker daemon.
 2. The Docker daemon pulled the "hello-world" image from the Docker Hub.
    (amd64)
 3. The Docker daemon created a new container from that image which runs the
    executable that produces the output you are currently reading.
 4. The Docker daemon streamed that output to the Docker client, which sent it
    to your terminal.

To try something more ambitious, you can run an Ubuntu container with:
 $ docker run -it ubuntu bash

Share images, automate workflows, and more with a free Docker ID:
 https://hub.docker.com/

For more examples and ideas, visit:
 https://docs.docker.com/get-started/
```

So, I know how to circumvent the problem, until it gets fixed in Arch, I still wonder, and that's why I open this issue:

Are you aware of the implications of including `"net.ipv4.ping_group_range=0 0"` in the default `containers.conf` ?

Is this something to be handled by the Linux distributions shipping `podman` or should this rather be handled by you?

**Output of `podman version`:**

```
Version:      2.1.1
API Version:  2.0.0
Go Version:   go1.15.5
Git Commit:   9f6d6ba0b314d86521b66183c9ce48eaa2da1de2
Built:        Tue Nov 24 18:53:00 2020
OS/Arch:      linux/amd64
```

**Output of `podman info --debug`:**

```
host:
  arch: amd64
  buildahVersion: 1.16.1
  cgroupManager: systemd
  cgroupVersion: v1
  conmon:
    package: Unknown
    path: /usr/bin/conmon
    version: 'conmon version 2.0.21, commit: 35a2fa83022e56e18af7e6a865ba5d7165fa2a4a'
  cpus: 8
  distribution:
    distribution: arch
    version: unknown
  eventLogger: journald
  hostname: jonny-arch-pc
  idMappings:
    gidmap: null
    uidmap: null
  kernel: 5.4.78-1-lts
  linkmode: dynamic
  memFree: 9825181696
  memTotal: 16746680320
  ociRuntime:
    name: runc
    package: Unknown
    path: /usr/bin/runc
    version: |-
      runc version 1.0.0-rc92
      commit: ff819c7e9184c13b7c2607fe6c30ae19403a7aff
      spec: 1.0.2-dev
  os: linux
  remoteSocket:
    path: /run/podman/podman.sock
  rootless: false
  slirp4netns:
    executable: ""
    package: ""
    version: ""
  swapFree: 0
  swapTotal: 0
  uptime: 1h 50m 9.3s (Approximately 0.04 days)
registries: {}
store:
  configFile: /etc/containers/storage.conf
  containerStore:
    number: 3
    paused: 0
    running: 0
    stopped: 3
  graphDriverName: overlay
  graphOptions: {}
  graphRoot: /var/lib/containers/storage
  graphStatus:
    Backing Filesystem: extfs
    Native Overlay Diff: "false"
    Supports d_type: "true"
    Using metacopy: "true"
  imageStore:
    number: 1
  runRoot: /var/run/containers/storage
  volumePath: /var/lib/containers/storage/volumes
version:
  APIVersion: 2.0.0
  Built: 1606240380
  BuiltTime: Tue Nov 24 18:53:00 2020
  GitCommit: 9f6d6ba0b314d86521b66183c9ce48eaa2da1de2
  GoVersion: go1.15.5
  OsArch: linux/amd64
  Version: 2.1.1
```

**Package info (e.g. output of `rpm -q podman` or `apt list podman`):**

```
[jonny@jonny-arch-pc ~]$ pacman -Qi podman
Name            : podman
Version         : 2.1.1-2
Description     : Tool and library for running OCI-based containers in pods
Architecture    : x86_64
URL             : https://github.com/containers/libpod
Licenses        : Apache
Groups          : None
Provides        : None
Depends On      : cni-plugins  conmon  containers-common  device-mapper
                  iptables  libseccomp  runc  slirp4netns  libsystemd
                  fuse-overlayfs  libgpgme.so=11-64
Optional Deps   : podman-docker: for Docker-compatible CLI
                  btrfs-progs: support btrfs backend devices [installed]
                  catatonit: --init flag support
                  crun: support for unified cgroupsv2
Required By     : None
Optional For    : None
Conflicts With  : None
Replaces        : None
Installed Size  : 77.94 MiB
Packager        : David Runge <dvzrv@archlinux.org>
Build Date      : Tue 24 Nov 2020 06:53:00 PM CET
Install Date    : Wed 25 Nov 2020 06:16:20 PM CET
Install Reason  : Explicitly installed
Install Script  : No
Validated By    : Signature
```

**Have you tested with the latest version of Podman and have you checked the Podman Troubleshooting Guide?**

Yes

**Comment by rhatdan** [comment]
We need to figure out why this is not supported on ArchLinux?

Is this a runc issue?

Error: sysctl "net.ipv4.ping_group_range" is not allowed in the hosts network namespace: OCI runtime error

@giuseppe WDYT?

**Comment by rhatdan** [comment]
@polygamma Any chance you could try this with crun?

Perhaps we should ignore these default settings, when the user sets the --net=host.

**Comment by rhatdan** [comment]
What version of podman is this happening with?

The code looks like this use case is supposed to be handled, https://github.com/containers/podman/blob/master/pkg/specgen/generate/security.go#L196

This is working fine for me in podman-2.2.0-0.4.rc1.fc33.x86_64

**Comment by rhatdan** [comment]
BTW in your homedir, if you setup /etc/containers/containers.conf or $HOME/.config/containers/containers.conf 

with

[containers]
default_sysctls = [
]

It should remove the sysctl.

**Comment by polygamma** [comment]
> What version of podman is this happening with?
> 
> The code looks like this use case is supposed to be handled, https://github.com/containers/podman/blob/master/pkg/specgen/generate/security.go#L196
> 
> This is working fine for me in podman-2.2.0-0.4.rc1.fc33.x86_64

I was using 2.1.1, but I just built 2.2.2-rc2 from source and that indeed fixes the problem

**Comment by dvzrv** [comment]
Hi! I did some of the packaging on the podman update when we introduced a containers-common package yesterday.

So, IIUC this can either be fixed by removing the currently present entry for `default_sysctl` during packaging or wait until 2.2.0 is released?
Or does the entry have to be removed anyways? If so I can open a pull request to remove it [here](https://github.com/containers/common/blob/master/pkg/config/containers.conf#L82). :)

**Comment by dvzrv** [comment]
Looking at the default file, I also see that a few `default_capabilities` are defined. These also cause issues though ([FS#68743](https://bugs.archlinux.org/task/68743)).

Checking on a fedora32 host I see that both `default_sysctls` and `default_capabilities` are commented by default. I can certainly also do that during packaging, but to serve as a sane default, I think the file should probably just have those commented (upstream), if possible.

**Comment by dvzrv** [comment]
@rhatdan this is related to https://github.com/containers/common/issues/364

**Comment by rhatdan** [comment]
The issue is a mismatch between the version of podman and containers.conf, I believe.  We can comment this out upstream, and switch the defaults in the code if you want.  My goal is to push forward the new defaults, but allow distributions to choose their defaults.  I believe the new version is the most secure.  BUT it has to have an updated version of Podman to work correctly.

**Comment by rhatdan** [comment]
In Fedora 33 we have the new containers.conf. We did not update fedora 32 with it, yet.  My preference would be to not update Fedora 32.

We are planning on releasing 2.2.0 early next week.  rc2 seems to be working well.

**Comment by dvzrv** [comment]
I understand. Maybe it would be most straight forward to provide the internal defaults (commented) in the default containers.conf.

Are there any operating system specific things to take care of to accommodate e.g. for the dropped capabilities with podman 2.2.0 or beyond? Would be great to sync up on that.

I'll keep the defaults commented until podman 2.2.0 is out.

**Comment by giuseppe** [comment]
> We need to figure out why this is not supported on ArchLinux?
> 
> Is this a runc issue?
> 
> Error: sysctl "net.ipv4.ping_group_range" is not allowed in the hosts network namespace: OCI runtime error
> 
> @giuseppe WDYT?

this is a correct error from the OCI runtime.  Podman should not set the `net.*` sysctls when using the host network namespace.

I think the commit that fixed it in Podman is: 0d70df119539d818224b0d014602aaad2bd1b95e

**Comment by rhatdan** [comment]
Since this is fixed in master and podman-2.2 I am closing this issue.

**Comment by jeffmccune** [comment]
Seems to be back on Debian 10

```
❯ sudo podman version
Version:      3.0.1
API Version:  3.0.0
Go Version:   go1.14
Built:        Wed Dec 31 16:00:00 1969
OS/Arch:      linux/amd64
```

```
Error: OCI runtime error: unable to start container d86bfe05117c2112793b1c374998460eeafdbaef6ea77832b76bd286b1396ec7: sysctl "net.ipv4.ping_group_range" is not allowed in the hosts network namespace
```

**Comment by jeffmccune** [comment]
Scratch that, after `podman rm mycontainer` and re-creating it, the error resolved itself with podman 3.0.1
